src/pages/base.js

```diff
@@ -16,8 +16,18 @@ class Page {
   visitWithParameters(data) {
     let url = this.url;
 
+    const additionalParams = [];
     for (const item of data.raw()) {
-      url = url.replace(`:${item[0]}`, item[1]);
+      if (url.indexOf(`:${item[0]}`) === -1) {
```

Comment:
That would be good to have this feature tested in our functional tests.

Here's an example
https://github.com/TheSoftwareHouse/Kakunin/blob/next/functional-tests/features/navigation/navigate_to_given_page.feature